src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,214 @@
+          <v-btn large outlined color="primary" id="start-cancel-btn" @click="onCancelClicked()">
+            <span>Cancel</span>
+          </v-btn>
+        </div>
+
+        <span class="mt-n2 mr-n3 float-right">
+          <!-- Correct changes button -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && !isCorrected" text color="primary"
+            id="start-changes-btn" @click="onChangeClicked()">
+            <v-icon small color="primary">mdi-pencil</v-icon>
+            <span>{{ editLabel }}</span>
+          </v-btn>
+
+          <!-- Undo changes buttons -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && isCorrected"
+          text color="primary" id="start-undo-btn" @click="onUndoClicked()">
+            <v-icon small>mdi-undo</v-icon>
+            <span>Undo</span>
+          </v-btn>
+          <v-menu v-if="isCorrectionFiling && !onEditMode && isCorrected"
+            offset-y left nudge-bottom="4" v-model="dropdown">
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" id="start-menu-btn" v-on="on">
+                <v-icon>{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectedFilingDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionFilingDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedFilingDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
```

Comment:
See below comment first
StartDateMin is also in utc time however unlike below it didn't seem to effect the minimum start date however I thought it would be best practice to use geUTCFullYear() anyways

Reply:
This looks fine to me. It's the same code as in Create UI so if it's wrong we will now have twice as many opportunities to find the bug and it will simply need to be fixed in both UIs.